controllers/status.go

```diff
@@ -240,6 +241,49 @@ func (r *PerformanceProfileReconciler) getKubeletConditionsByProfile(profile *pe
 	return r.getDegradedConditions(conditionKubeletFailed, latestCondition.Message), nil
 }
 
+func (r *PerformanceProfileReconciler) getTunedConditionsByProfile(profile *performancev2.PerformanceProfile) ([]conditionsv1.Condition, error) {
+	tunedProfileList := &tunedv1.ProfileList{}
+	if err := r.List(context.TODO(), tunedProfileList); err != nil {
+		klog.Errorf("Cannot list Tuned Profiles to match with profile %q : %v", profile.Name, err)
+		return nil, err
+	}
+
+	message := bytes.Buffer{}
+	for _, tunedProfile := range tunedProfileList.Items {
+		isDegraded := false
+		isApplied := true
+		var pDegCondition *tunedv1.ProfileStatusCondition
+
+		for _, condition := range tunedProfile.Status.Conditions {
+			if (condition.Type == tunedv1.TunedDegraded) && condition.Status == corev1.ConditionTrue {
+				isDegraded = true
+				pDegCondition = &condition
+			}
+
+			if (condition.Type == tunedv1.TunedProfileApplied) && condition.Status == corev1.ConditionFalse {
+				isApplied = false
+			}
+		}
+		// We need both condition to exists,
+		// since there is a scenario where both Degraded & Applied condition are ture
+		if isDegraded == true && isApplied == false {
+			if len(pDegCondition.Reason) > 0 {
+				message.WriteString("Tuned " + tunedProfile.GetName() + " Degraded Reason: " + pDegCondition.Reason + ".\n")
```

Comment:
why a `bytes.Buffer` and not, say, just a plain `fmt.Sprintf` ?

Reply:
No special reason. Just followed the convention: https://github.com/openshift-kni/performance-addon-operators/blob/master/controllers/status.go#L199

Reply:
well, ok, let's change all these in a later PR